### ISSUE TITLE
LIPS command line `(read-line (current-input-port))` erases prompt

### ISSUE THREAD

**Issue by dragoncoder047** [jcubic/lips#253]
`(read-line (current-input-port))` is the way I've seen to get a line of input form the user in Scheme. Unfortunately, on LIPS, this doesn't work quite-right, it just erases the prompt. (I'm looking for something equivalent to Python's `input()` function.)

```fish
dc047:/home/dc047> npm list -g | grep lips
├── @jcubic/lips@1.0.0-beta.16
dc047:/home/dc047> lips --version
(command "lips")
(website "https://lips.js.org")
(languages scheme r5rs r7rs)
(encodings utf-8)
(scheme.srfi 0 1 2 4 6 8 10 22 23 26 46 69 98 111 156 176)
(release "{{VER}}")
(os.uname "linux" "6.2.0-1016-azure")
(os.env.LANG "C.UTF-8")
(os.env.TERM "xterm-256color")
(build.date "{{DATE}}")
dc047:/home/dc047> cat test.scm
(display "Enter a string: ")
; (wait 1000)
(display (concat "You entered: " (read-line (current-input-port))))
dc047:/home/dc047> lips test.scm
hello
You entered: hello                                                                                                                        
```

It's supposed to wind up with "Enter a string: hello" on the first line after running test.scm. If you uncomment the (wait) call, you can see that the prompt is actually printed, but when it gets to the (read-line), the line is cleared.

If there is a better way of getting input, please let me know.

**Comment by jcubic** [comment]
Thanks for the report, will try to check it out when I will have time.

**Comment by jcubic** [comment]
So the problem is because of the way stdin/stdout are created in REPL:

```javascript
var interp = Interpreter('repl', {
    stdin: InputPort(function() {
        return new Promise(function(resolve) {
            rl = readline.createInterface({
                input: process.stdin,
                output: process.stdout
            });
            rl.question('', function(data) {
                resolve(data);
                rl.close();
            });
        });
    }),
    // -------------------------------------------------------------------------
    stdout: OutputPort(function(x) {
        if (typeof x !== 'string') {
            x = this.get('repr')(x);
        }
        newline = !x.match(/\n$/);
        process.stdout.write(x);
    }),
```

NodeJS `readline.createInterface` clears everything that was printed in same line. I think that the quick fix would be to keep text before newline in variable and use it in `rl.question` instead of empty string.

**Comment by jcubic** [comment]
the fix seems to work fine. It's available on devel branch.

**Comment by dragoncoder047** [comment]
how do I `npm install -g` the devel branch? Do I just point to a specific commit?

**Comment by dragoncoder047** [comment]
nevermind, I figured it out.

**Comment by jcubic** [comment]
Just released version 1.0.0-beta.17 with a fix.
